Subject: [PATCH] update docker build
update docker build
update docker build
update docker build
update docker build
update docker build
update docker build
update docker build
update docker build
update docker build
update versions
---
Index: .github/workflows/sync-upstream-manual.yml
===================================================================
diff --git a/.github/workflows/sync-upstream-manual.yml b/.github/workflows/sync-upstream-manual.yml
new file mode 100644
--- /dev/null	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
+++ b/.github/workflows/sync-upstream-manual.yml	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
@@ -0,0 +1,19 @@
+name: Merge upstream main branch (manual)
+on: workflow_dispatch
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Merge upstream
+        run: |
+          git config --global user.name 'Marko Smej'
+          git config --global user.email 'markoftw@users.noreply.github.com'
+          git pull --unshallow  
+          git remote add upstream https://github.com/tailscale/tailscale.git
+          git fetch upstream
+          git checkout main
+          git merge --no-edit upstream/main
+          git push origin main
+          git push origin main --tags
+          git push origin --tags
Index: .github/workflows/sync-upstream.yml
===================================================================
diff --git a/.github/workflows/sync-upstream.yml b/.github/workflows/sync-upstream.yml
new file mode 100644
--- /dev/null	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
+++ b/.github/workflows/sync-upstream.yml	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
@@ -0,0 +1,21 @@
+name: Merge upstream main branch
+on:
+  schedule:
+    - cron:  '0 10 * * *'
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Merge upstream
+        run: |
+          git config --global user.name 'Marko Smej'
+          git config --global user.email 'markoftw@users.noreply.github.com'
+          git pull --unshallow  
+          git remote add upstream https://github.com/tailscale/tailscale.git
+          git fetch upstream
+          git checkout main
+          git merge --no-edit upstream/main
+          git push origin main
+          git push origin main --tags
+          git push origin --tags
Index: .gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.gitignore b/.gitignore
--- a/.gitignore	(revision ed843e643f9f311fc00808253abb4e829adf7af3)
+++ b/.gitignore	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
@@ -43,3 +43,5 @@
 
 /gocross
 /dist
+
+.idea/
Index: entrypoint.sh
===================================================================
diff --git a/entrypoint.sh b/entrypoint.sh
new file mode 100644
--- /dev/null	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
+++ b/entrypoint.sh	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
@@ -0,0 +1,10 @@
+/usr/local/bin/tailscaled --tun=userspace-networking &
+until /usr/local/bin/tailscale up --authkey=${AUTH_KEY} --hostname=${NAME} --advertise-routes=${ROUTES} --accept-dns=false ${EXTRA_ARGS}
+do
+    sleep 0.1
+done
+
+while pgrep tailscaled >/dev/null
+do
+    sleep 1
+done
Index: .github/workflows/docker-file-build.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.github/workflows/docker-file-build.yml b/.github/workflows/docker-file-build.yml
--- a/.github/workflows/docker-file-build.yml	(revision 1c59fe5e19df2e8d750cacaa29317ba32d86d8fc)
+++ b/.github/workflows/docker-file-build.yml	(revision 39a98c5523c44179b0b6ca93a29df25bd43f4241)
@@ -3,13 +3,48 @@
   push:
     branches:
     - main
-  pull_request:
-    branches:
-      - "*"
+#  pull_request:
+#    branches:
+#      - "*"
+
 jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    - name: "Build Docker image"
-      run: docker build .
+
+    - name: Get current version
+      id: tailscale_version
+      run: |
+        app_version=$(cat VERSION.txt)
+        
+        echo $app_version
+        echo "app_version=$app_version" >> $GITHUB_OUTPUT
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+    - name: Set values for tagging Docker image
+      id: tags_meta
+      uses: docker/metadata-action@v5
+      with:
+        images: "markoftw/tailscale"
+        tags: |
+          type=raw,value=${{ steps.tailscale_version.outputs.app_version }}
+          type=raw,value=latest
+
+    - name: Build and Push Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: ./markoftw.Dockerfile
+        platforms: linux/amd64
+        push: true
+        tags: ${{ steps.tags_meta.outputs.tags }}
+#        build-args:
+#          VERSION_SHORT=${{ steps.tailscale_version.outputs.app_version }}
+#          VERSION_LONG=${{ steps.tailscale_version.outputs.app_version }}
+#          VERSION_GIT_HASH=${{ github.sha }}
Index: markoftw.Dockerfile
===================================================================
diff --git a/markoftw.Dockerfile b/markoftw.Dockerfile
new file mode 100644
--- /dev/null	(revision 39a98c5523c44179b0b6ca93a29df25bd43f4241)
+++ b/markoftw.Dockerfile	(revision 39a98c5523c44179b0b6ca93a29df25bd43f4241)
@@ -0,0 +1,34 @@
+FROM golang:1.22-alpine AS build-env
+
+WORKDIR /go/src/tailscale
+
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . .
+
+# see build_docker.sh
+ARG VERSION_LONG=""
+ENV VERSION_LONG=$VERSION_LONG
+ARG VERSION_SHORT=""
+ENV VERSION_SHORT=$VERSION_SHORT
+ARG VERSION_GIT_HASH=""
+ENV VERSION_GIT_HASH=$VERSION_GIT_HASH
+
+RUN go install -tags=xversion -ldflags="\
+      -X tailscale.com/version.longStamp=$VERSION_LONG \
+      -X tailscale.com/version.shortStamp=$VERSION_SHORT \
+      -X tailscale.com/version.gitCommitStamp=$VERSION_GIT_HASH" \
+      -v ./cmd/tailscale ./cmd/tailscaled ./cmd/containerboot
+
+FROM alpine:3.18
+
+RUN apk add --no-cache ca-certificates iptables iproute2 ip6tables iputils
+
+COPY --from=build-env /go/bin/* /usr/local/bin/
+
+COPY entrypoint.sh /usr/local/bin/entrypoint.sh
+
+RUN chmod a+x /usr/local/bin/entrypoint.sh
+
+CMD /usr/local/bin/entrypoint.sh
\ No newline at end of file
